src/scenes/ProgressReports/EditForm/Steps/ProgressStep/ProgressStep.tsx

```diff
@@ -0,0 +1,239 @@
+      })}
+    >
+      <Grid container spacing={2} sx={{ mb: 4 }}>
+        {tabIndex ? (
+          <Grid item md={6}>
+            <ProgressSummaryCard
+              loading={!report}
+              summary={report.cumulativeSummary ?? null}
+              sx={{ height: 1 }}
+            />
+          </Grid>
+        ) : null}
+        <Grid container item md={tabIndex ? 6 : 12} spacing={2}>
+          {!report.reportFile.value && (
+            <Grid item md={6} sx={{ display: 'flex', alignItems: 'center' }}>
+              <Typography variant="body2">
+                (TENTATIVE COPY) Please upload the PnP for this reporting
+                period. The progress data will populate the charts below.
+              </Typography>
+            </Grid>
+          )}
+          <Grid item md={6} justifyItems="end">
+            <ProgressReportCard
+              progressReport={report}
+              disableIcon
+              onUpload={({ files }) => setUploading(files)}
+            />
+          </Grid>
+        </Grid>
+      </Grid>
+
+      {categories.map((category) => (
+        <Box key={category} sx={{ mb: 4 }}>
+          <EditableProductTable
+            products={forTable[category]}
+            category={category}
+            update={update}
+            report={report}
+            tab={tab}
+            setTab={setTab}
+          />
+        </Box>
+      ))}
+
+      <UpdatePeriodicReportDialog
+        {...dialogState}
+        report={{ ...report, reportFile: upload }}
+        editFields={[
+          'receivedDate',
+          ...(upload && upload.length > 0 ? ['reportFile' as const] : []),
+        ]}
+      />
+    </div>
+  );
+};
+
+const EditableProductTable = ({
+  category,
+  products = [],
+  update,
+  report,
+  tab,
+  setTab,
+}: {
+  category: string;
+  products?: ProgressReportProgressFragment[];
+  update: any;
+  report: ProgressReportEditFragment;
+  tab: typeof variants[number];
+  setTab: (tab: typeof variants[number]) => void;
+}) => (
+  <ProductTable
+    category={category}
+    products={products}
+    pagination
+    header={() => (
+      <div style={{ display: 'flex', justifyContent: 'space-between' }}>
+        <VariantsToggle
+          setTab={(value) => setTab(variants[value] ?? variants[0])}
+          tab={variants.indexOf(tab)}
+        />
+      </div>
+    )}
+    editMode="row"
+    onRowEditStop={(fields) => {
+      void update({
+        variables: {
+          input: {
+            productId: fields.id.toString(),
+            reportId: report.id,
+            variant: tab,
+            steps: [
+              ...fields.row.data.steps.map((step: StepProgressFragment) => ({
+                completed: parseFloat(fields.row[step.step]),
+                percentDone: parseFloat(fields.row[step.step]),
+                step: step.step,
+              })),
+            ],
+          },
+        },
+      });
+    }}
+  />
+);
+
+const VariantsToggle = ({
```

Comment:
This should accept a variant object list from the API to populate this toggle dynamically.